TreasureHunt/Assets/2.Scripts/Hunter/Buttons/MenuButtonController.cs

```diff
@@ -0,0 +1,31 @@
+﻿using UnityEngine;
+using System.Collections;
+using UnityEngine.UI;
+
+public class MenuButtonController : MonoBehaviour {
+
+	bool flag = false;
+	public GameObject target;
+
+	Button myButton;
+
+	void Awake(){
+		myButton = GetComponent<Button> ();
+		myButton.onClick.AddListener (GotIt);
+	}
+
+	void Start(){
+		target.SetActive (false);
+	}
+
+	void GotIt(){
+		if (flag) {
```

Comment:
This code seems slightly redundant. I think you can do without the flags.

Notice that GameObject has an [activeSelf](https://docs.unity3d.com/ScriptReference/GameObject-activeSelf.html) field.

What about the following:

```
void GotIt() {
    target.SetActive(!target.activeSelf);
}
```